espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/MobileBackdoorMethod.java

```diff
@@ -0,0 +1,49 @@
+package io.appium.espressoserver.lib.model;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import io.appium.espressoserver.lib.helpers.BackdoorUtils;
+
+@SuppressWarnings("unused")
+public class MobileBackdoorMethod {
+    private String name;
+
+    @Nullable
+    private List<BackdoorMethodArg> args;
+
+    public String getName() {
+        return name;
+    }
+
+    @Nullable
+    public List<BackdoorMethodArg> getRawArgs() {
+        if (args == null) {
+            return Collections.emptyList();
+        } else {
+            return args;
+        }
+
+    }
+
+    public Class<?>[] getArgumentTypes() {
+        List<BackdoorMethodArg> rawArgs = getRawArgs();
+        Class<?>[] types = new Class<?>[rawArgs.size()];
+        for (int i = 0; i < rawArgs.size(); i++) {
+            types[i] = BackdoorUtils.parseType(getRawArgs().get(i).getType());
+        }
+        return types;
+    }
+
+    public List<Object> getArguments() {
```

Comment:
is there any reason to extract types as array and arguments as list? Cannot it be the same type?

Reply:
Because Java.lang.Class.getMethod() expects array